From 0fb9b841ac69a51b7443e432006806028f717872 Mon Sep 17 00:00:00 2001
From: Vinitha V Pillai <vinitha.pillai@nxp.com>
Date: Tue, 27 Feb 2018 12:57:31 +0530
Subject: [PATCH 005/107] LS2088ARDB: Secure Boot: Fix fall back option failure

Fix NOR, SD and QSPI fallback option in case of secure boot failure.

Signed-off-by: Vinitha V Pillai <vinitha.pillai@nxp.com>
Reviewed-by: York Sun <york.sun@nxp.com>
---
 include/configs/ls2080ardb.h |   14 +++++++-------
 1 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/include/configs/ls2080ardb.h b/include/configs/ls2080ardb.h
index 3f9794f..299362c 100644
--- a/include/configs/ls2080ardb.h
+++ b/include/configs/ls2080ardb.h
@@ -372,7 +372,7 @@ unsigned long get_board_sys_clk(void);
 	"fdt_high=0xa0000000\0"			\
 	"initrd_high=0xffffffffffffffff\0"	\
 	"fdt_addr=0x64f00000\0"			\
-	"kernel_addr=0x65000000\0"		\
+	"kernel_addr=0x581000000\0"		\
 	"kernel_start=0x1000000\0"		\
 	"kernelheader_start=0x800000\0"		\
 	"scriptaddr=0x80000000\0"		\
@@ -442,8 +442,8 @@ unsigned long get_board_sys_clk(void);
 			"&& esbc_validate 0x20780000; "			\
 			"env exists mcinitcmd && "			\
 			"fsl_mc lazyapply dpl 0x20d00000; "		\
-			"run distro_bootcmd;env exists secureboot "	\
-			" && esbc_halt;run qspi_bootcmd; "
+			"run distro_bootcmd;run qspi_bootcmd; "		\
+			"env exists secureboot && esbc_halt;"
 #elif defined(CONFIG_SD_BOOT)
 /* Try to boot an on-SD kernel first, then do normal distro boot */
 #define CONFIG_BOOTCOMMAND						\
@@ -453,16 +453,16 @@ unsigned long get_board_sys_clk(void);
 			"env exists mcinitcmd && run mcinitcmd "	\
 			"&& mmc read 0x88000000 0x6800 0x800 "		\
 			"&& fsl_mc lazyapply dpl 0x88000000; "		\
-			"run distro_bootcmd;env exists secureboot "	\
-			"&& esbc_halt;run sd_bootcmd;"
+			"run distro_bootcmd;run sd_bootcmd; "		\
+			"env exists secureboot && esbc_halt;"
 #else
 /* Try to boot an on-NOR kernel first, then do normal distro boot */
 #define CONFIG_BOOTCOMMAND						\
 			"env exists mcinitcmd && env exists secureboot "\
 			"&& esbc_validate 0x580780000; env exists mcinitcmd "\
 			"&& fsl_mc lazyapply dpl 0x580d00000;"		\
-			"run distro_bootcmd; env exists secureboot "	\
-			"&& esbc_halt; run nor_bootcmd;"
+			"run distro_bootcmd;run nor_bootcmd; "		\
+			"env exists secureboot && esbc_halt;"
 #endif
 
 /* MAC/PHY configuration */
-- 
1.7.1

